languageserver/integration/codelenses.go

```diff
@@ -159,72 +257,162 @@ func (i *FlowIntegration) entryPointActions(
-			)
-		}
+	for index, argumentList := range argumentLists {
+		var title string
 
-		encodedArgumentList := make([]string, len(argumentList))
-		for i, argument := range argumentList {
+		var encodedArgumentList []string
+		for _, argument := range argumentList {
 			encodedArgument, err := jsoncdc.Encode(argument)
 			if err != nil {
 				return nil, err
 			}
-			encodedArgumentList[i] = string(encodedArgument)
+			noSuffix := strings.TrimSuffix(string(encodedArgument), "\n")
+			encodedArgumentList = append(encodedArgumentList, noSuffix)
 		}
 
-		codeLenses[i] = &protocol.CodeLens{
-			Range: conversion.ASTToProtocolRange(
-				position,
-				position,
-			),
-			Command: &protocol.Command{
-				Title:     formattedTitle,
-				Command:   command,
-				Arguments: []interface{}{uri, encodedArgumentList},
-			},
+		switch entryPointInfo.kind {
+
+		case entryPointKindScript:
+			if len(argumentList) > 0 {
+				title = fmt.Sprintf(
+					"%s Execute script with %s",
+					prefixOK,
+					entryPointInfo.pragmaArgumentStrings[index],
+				)
+			} else {
+				title = fmt.Sprintf(
+					"%s Execute script",
+					prefixOK,
+				)
+			}
+
+			argsJSON := convertListToCadenceJSON(encodedArgumentList)
+
+			codeLens := &protocol.CodeLens{
+				Range: codelensRange,
+				Command: &protocol.Command{
+					Title:     title,
+					Command:   CommandExecuteScript,
+					Arguments: []interface{}{uri, argsJSON},
+				},
+			}
+			codeLenses = append(codeLenses, codeLens)
+
+		case entryPointKindTransaction:
+			for _, signers := range signersList {
+				if signers[0] == "no-signers" {
+					signers = []string{i.activeAccount.Name}
+				}
+
+				var emptyAccounts []string
+				var resolvedAccounts []flow.Address
+				for _, signer := range signers {
+					resolvedAddress, _ := i.getAccountAddress(signer)
+					resolvedAccounts = append(resolvedAccounts, resolvedAddress)
+					if addressNotExist(resolvedAddress) {
+						emptyAccounts = append(emptyAccounts, signer)
+					}
+				}
+				var codeLens *protocol.CodeLens
+				if len(emptyAccounts) > 0 {
+					accountsNumeric := "account"
+					if len(emptyAccounts) > 1 {
+						accountsNumeric = "accounts"
+					}
+					title = fmt.Sprintf("%s Specified %s %s does not exist",
+						prefixError,
+						accountsNumeric,
+						common.EnumerateWords(emptyAccounts, "and"),
+					)
+					codeLens = &protocol.CodeLens{
+						Range: codelensRange,
+						Command: &protocol.Command{
+							Title: title,
+						},
+					}
+				} else {
+					if len(argumentList) > 0 {
+						title = fmt.Sprintf(
+							"%s Send with %s signed by %s",
+							prefixOK,
+							entryPointInfo.pragmaArgumentStrings[index],
+							common.EnumerateWords(signers, "and"),
+						)
+					} else {
+						title = fmt.Sprintf(
+							"%s Send signed by %s",
+							prefixOK,
+							common.EnumerateWords(signers, "and"),
+						)
+					}
+
+					argsJSON := convertListToCadenceJSON(encodedArgumentList)
```

Comment:
Same as above, instead of preparing the arguments separately and then using string formatting and replacing characters to produce the final JSON (in `convertListToCadenceJSON`), consider JSON marshaling a Go slice with the arguments